tests/integration-tests/tests/byos/test_byos.py

```diff
@@ -9,19 +9,172 @@
+            launch_templates_list.append(compute_resource.get("LaunchTemplate").get("Id"))
+    assert_that(launch_templates_list).is_length(3)
+
+    return launch_templates_list
+
+
+def _start_compute_node(region, cluster, command_executor):
+    """Start and wait compute node to finish setup"""
+    # Get one launch template for compute node
+    lt = _get_launch_templates(command_executor)[0]
+    run_instance_command = (
+        f"aws ec2 run-instances --region {region}"
+        f" --instance-initiated-shutdown-behavior terminate"
+        f" --count 1"
+        f" --launch-template LaunchTemplateId={lt}"
+    )
+    command_executor.run_remote_command(run_instance_command)
+    # Wait instance to start
+    time.sleep(3)
+    compute_nodes = cluster.describe_cluster_instances(node_type="Compute")
+    assert_that(compute_nodes).is_length(1)
+    compute_node = compute_nodes[0]
+    instance_id = compute_node.get("instanceId")
+    # Wait instance to go running
+    _wait_instance_running(region, [instance_id])
+    # Wait instance to complete cloud-init
+    _wait_compute_cloudinit_done(command_executor, compute_node)
+
+    return compute_node
+
+
+def _wait_instance_running(region, instance_ids):
+    """Wait EC2 instance to go running"""
+    logging.info(f"Waiting for {instance_ids} to be running")
+    boto3.client("ec2", region_name=region).get_waiter("instance_running").wait(
+        InstanceIds=instance_ids, WaiterConfig={"Delay": 60, "MaxAttempts": 5}
+    )
+
+
+@retry(wait_fixed=seconds(10), stop_max_delay=minutes(3))
+def _wait_compute_cloudinit_done(command_executor, compute_node):
+    """Wait till cloud-init complete on a given compute node"""
+    compute_node_private_ip = compute_node.get("privateIpAddress")
+    compute_cloudinit_status_output = command_executor.run_remote_command(
+        f"ssh -q {compute_node_private_ip} sudo cloud-init status"
+    ).stdout
+    assert_that(compute_cloudinit_status_output).contains("status: done")
+
+
+def _test_event_handler_execution(command_executor, compute_node):
+    """Test event handler execution and environment"""
+    head_byos_log_output = command_executor.run_remote_command(f"cat {BYOS_LOG_PATH}").stdout
+    for event in ["HeadInit", "HeadConfigure", "HeadFinalize"]:
+        assert_that(head_byos_log_output).contains(f"[{event}] - INFO: {event} executed")
+        _test_event_handler_environment(event, head_byos_log_output)
+
+    compute_node_private_ip = compute_node.get("privateIpAddress")
+    compute_byos_log_output = command_executor.run_remote_command(
+        f"ssh -q {compute_node_private_ip} cat {BYOS_LOG_PATH}"
+    ).stdout
+    for event in ["ComputeInit", "ComputeConfigure", "ComputeFinalize"]:
+        assert_that(compute_byos_log_output).contains(f"[{event}] - INFO: {event} executed")
+        _test_event_handler_environment(event, compute_byos_log_output)
+
+
+def _test_event_handler_environment(event, log_output):
```

Comment:
shall we assert the actual value of these variables to make sure we set them correctly?

Reply:
added